AI mode implementation

e-p-armstrong opened the pull request.

OnAirNetlines committed:
(#1) Import files from agent-stuff that I use in new system

OnAirNetlines committed:
(#1) Implement AI mode feed in a clean way

OnAirNetlines committed:
(#1) Simplify the existing agent-stuff files for new system

The prompt change was required because posts without text but with
media does contain the author, so we can't say the AI will not know
the author of a post.

OnAirNetlines committed:
(#1) Integrate the AI mode feed into Bluesky code

e-p-armstrong committed:
implement personality edit

e-p-armstrong committed:
save personality automatic update basics

e-p-armstrong committed:
fix problems with update display

e-p-armstrong committed:
save functional

e-p-armstrong committed:
merge upstream

e-p-armstrong committed:
fix ijmports

e-p-armstrong committed:
save

mcstoer commented on package-lock.json:
Since the project is using yarn, could you please remove this `package-lock.json` file?

mcstoer commented on package.json:
```diff
@@ -248,7 +251,7 @@
     "babel-plugin-react-compiler": "^19.1.0-rc.1",
     "babel-preset-expo": "~13.1.11",
     "eslint": "^8.19.0",
-    "eslint-plugin-bsky-internal": "link:./eslint",
```
The upstream repository uses `link`. Could you please revert this change to help keep consistency?

mcstoer commented on src/lib/llm-feed/ai-merged-feed/AIMergedFeed.ts:
```diff
@@ -0,0 +1,116 @@
+import { AppBskyFeedDefs } from '@atproto/api';
```
For the imports, they should be auto-sorted and use prefixed imports when possible as according to the eslint rules. Would you mind running the auto-fix for this and the other files affected?

mcstoer commented on src/lib/llm-feed/ai-merged-feed/ShellAIFeedAPI.ts:
```diff
@@ -0,0 +1,183 @@
+  private async initializeRealFeed(): Promise<void> {
+    try {
+      // 1. Fetch user profile data
+      const userProfile = await getCurrentUserProfile(this.agent);
```
If the user hasn't logged in yet, this causes the app to error out on load. Could you please fix this?

mcstoer commented on src/lib/llm-feed/feed-curator.ts:
```diff
@@ -0,0 +1,157 @@
+  ): Promise<number[]> {
+    try {
+      // Log inputs
+      console.log('==== LLM FEED CURATION STARTED ====');
```
The upstream repository most often uses a logger for logging instead of `console.log`. Could you please switch to the logger to help keep consistency?

mcstoer commented on src/lib/llm-feed/feed-curator.ts:
```diff
@@ -0,0 +1,157 @@
+      
+      const chatCompletion = await openai.chat.completions.create({
+        messages: promptMessages as any,
+        model: 'mistralai/Mistral-Small-24B-Instruct-2501',
```
Would it be possible to make the model name configurable?

mcstoer commented on src/lib/llm-feed/feed-curator.ts:
```diff
@@ -0,0 +1,157 @@
+  private apiKey: string
+  private baseURL: string
+  private openai: OpenAI | null = null
+  private readonly defaultMaxPosts = 50;
```
Some of these variables are unused. Are they intended for specific future use?

mcstoer commented:
A `Failed to compile.` TypeError: [BABEL] occurs that can be reproduced by the following steps.
1. Checkout the `main` branch and run `yarn`
2. Checkout this branch and run `yarn`
3. Run `yarn web`

e-p-armstrong committed:
fix import types

e-p-armstrong committed:
Configurable api settings

e-p-armstrong committed:
fix update text color

e-p-armstrong committed:
save import fixes and other PR issues

e-p-armstrong committed:
save fix to model configuration -- no longer uses env

e-p-armstrong commented:
UPDATE -- I implemented things that should hnave addressed your comments but in trying to get it onto GitHub properly git problems have causedd this to get worse. I have the original changes backed up and will sort this out sometime early next week.

e-p-armstrong committed:
fix unused

e-p-armstrong committed:
merge finally

e-p-armstrong committed:
second merge

e-p-armstrong commented:
@mcstoer After going through git hell; changes made to address your comments, and merge conflicts resolved

mcstoer commented:
The commits from the Bluesky team are complicating the PR review. I tried removing them by updating the verusky branch, but was unsuccessful. Could you please clean up these commits?

On top of that, the PR contains several unrelated changes to AI feed implementation, which will risk creating conflicts with the main Bluesky repository. These changes are primarily adding `type` to the import statements. Can you please revert those changes?
